File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    global answer
    answer = 0
    
    def dfs(sum, idx):
        global answer
        if idx == len(numbers):
            if sum == target:
                answer += 1
            return
        
        dfs(sum + numbers[idx], idx+1)
        dfs(sum - numbers[idx], idx+1)
        
    dfs(0, 0)
    return answer

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.js
let nums = [];
let targetNumber = 0;
let count = 0;

function solution(numbers, target) {
    var answer = 0;
    nums = numbers;
    targetNumber = target;

    dfs(0, 0);
    return count;
}

function dfs(idx, sum) {
    if (idx == nums.length) {
        if (sum == targetNumber) {
            count++;
        }
        return;
    }

    dfs(idx + 1, sum + (-1 * nums[idx]));
    dfs(idx + 1, sum + (nums[idx]));
}